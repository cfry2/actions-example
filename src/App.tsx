import {useState} from 'react'
import './App.css'
import Input from './components/input'
import Button from './components/button'
import { useSelector, useDispatch } from 'react-redux'
import { sum } from './reducers/sumSlice'
import type { RootState } from './store'

function App() {
  const dispatch = useDispatch()
  const sumValue = useSelector((state: RootState) => state.sum)
  const [one, setOne] = useState('')
  const [two, setTwo] = useState('')

  const sumTwoNumbers = () => {
    dispatch(sum({
      one,
      two
    }))
  }

  return (
    <>
      <h1>Sum two numbers together</h1>
      <div>
      <Input id="one" label="one" value={one} onChange={(value) => setOne(value)} />
      </div>
      <div>
      <Input id="one" label="two" value={two} onChange={(value) => setTwo(value)} />
      </div>
      <Button onClick={sumTwoNumbers}/>
      <h2>value:</h2>
      <div>{sumValue.value}</div>
    </>
  )
}

export default App
