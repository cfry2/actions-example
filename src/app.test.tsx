import App from "./App";
import { test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Theme } from "@radix-ui/themes";
import { Provider } from "react-redux";
import userEvent from "@testing-library/user-event";

const user = userEvent.setup()
const store = configureStore([])({sum: {value: 0}});

test("handles actions", async () => {
   render(
    <Provider store={store}>
      <Theme>
        <App />
      </Theme>
    </Provider>
  );

  await user.type(screen.getByPlaceholderText('one'), "1")
  await user.type(screen.getByPlaceholderText('two'), "1")
  await user.click(screen.getByText('Go!'))
  expect(store.getActions()[0].type).toEqual('todos/sum')
});
