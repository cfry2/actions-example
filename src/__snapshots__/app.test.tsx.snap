// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handles actions 1`] = `
<body>
  <div>
    <style>
      
:root, .light, .light-theme { --color-page-background: white; }
.dark, .dark-theme { --color-page-background: var(--slate-1); }
body { background-color: var(--color-page-background); }

    </style>
    <div
      class="radix-themes"
      data-accent-color="indigo"
      data-gray-color="slate"
      data-has-background="false"
      data-is-root-theme="true"
      data-panel-background="translucent"
      data-radius="medium"
      data-scaling="100%"
    >
      <div
        style="max-width: 1100px; margin: 0px auto 0px auto;"
      >
        <div
          class="rt-Box rt-r-pb-6"
        >
          <h1
            class="rt-Heading rt-r-size-9 rt-r-weight-bold"
          >
            Sum two numbers together!
          </h1>
        </div>
        <div
          class="rt-Box rt-r-pb-6"
        >
          <div
            class="rt-Flex rt-r-display-flex rt-r-jc-center rt-r-gap-3"
          >
            <div
              class="rt-TextFieldRoot"
            >
              <input
                class="rt-TextFieldInput rt-r-size-2 rt-variant-surface"
                id="one"
                placeholder="one"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="rt-TextFieldChrome"
              />
            </div>
            <div
              class="rt-TextFieldRoot"
            >
              <input
                class="rt-TextFieldInput rt-r-size-2 rt-variant-surface"
                id="one"
                placeholder="two"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="rt-TextFieldChrome"
              />
            </div>
          </div>
        </div>
        <div
          class="rt-Box rt-r-pb-6"
        >
          <button
            class="rt-reset rt-BaseButton rt-Button rt-r-size-2 rt-variant-solid"
            type="button"
          >
            Go!
          </button>
        </div>
        <h2
          class="rt-Heading rt-r-size-8 rt-r-weight-bold"
        >
          value:
        </h2>
        <div>
          0
        </div>
      </div>
    </div>
  </div>
</body>
`;
